src/components/Activity.tsx

```diff
@@ -0,0 +1,150 @@
+import React from 'react';
+import { FileIcon } from 'react-file-utils';
+import { EnrichedActivity } from 'getstream';
+
+import { UserBar } from './UserBar';
+import { Card, CardProps } from './Card';
+import { Audio } from './Audio';
+import { Video } from './Video';
+import { Gallery } from './Gallery';
+import {
+  smartRender,
+  sanitizeURL,
+  userOrDefault,
+  textRenderer,
+  humanizeTimestamp,
+  ElementOrComponentOrLiteralType,
+  UserOrDefaultReturnType,
+} from '../utils';
+import { useTranslationContext } from '../Context';
+import { DefaultAT, DefaultUT } from '../Context/StreamApp';
+
+type WordClickHandler = (word: string) => void;
+
+export type ActivityProps<UT extends DefaultUT = DefaultUT, AT extends DefaultAT = DefaultAT> = {
+  activity: EnrichedActivity<UT, AT> & { object: { data: CardProps } | string };
+  icon?: string;
+  /** Handler for any routing you may do on clicks on Hashtags */
+  onClickHashtag?: WordClickHandler;
+  /** Handler for any routing you may do on clicks on Mentions */
+  onClickMention?: WordClickHandler;
+  onClickUser?: (user: UserOrDefaultReturnType<UT>) => void;
+  sub?: string;
+} & Partial<Record<'Header' | 'HeaderRight' | 'Footer' | 'Content', ElementOrComponentOrLiteralType>>;
+
+type DefaultActivityHeaderProps<UT extends DefaultUT = DefaultUT, AT extends DefaultAT = DefaultAT> = Pick<
+  ActivityProps<UT, AT>,
+  'activity' | 'HeaderRight' | 'icon' | 'onClickUser'
+>;
+type DefaultActivityContentProps<UT extends DefaultUT = DefaultUT, AT extends DefaultAT = DefaultAT> = Pick<
+  ActivityProps<UT, AT>,
+  'activity' | 'onClickMention' | 'onClickHashtag'
+>;
+
+const DefaultActivityHeader = <UT extends DefaultUT = DefaultUT, AT extends DefaultAT = DefaultAT>({
```

Comment:
since this is just a wrapper around Userbar, should we refactor it into its own file/component?